### PR TITLE
In work type modal, update options for Mixed Materials.

### LINKED ISSUE #2275: In work type modal, update options for "Mixed Materials"
When a user selects "Mixed Materials" in the work type modal, update the text that displays in red

from: Select at least two terms below:
to: Select at least two terms from any of the options below:

Also, the "See more options" section should be expanded by default for the "Mixed Materials" work type (but not for any other work type).

Remove the javascript that collapses the options. There should no longer be the "see more/fewer options"

Screen Shot 2022-06-08 at 12 59 46 PM

### PR DESCRIPTION
**Pull request by justinlittman**
closes #2275

## How was this change tested? 🤨

⚡ ⚠ If this change involves consuming from or writing to another service (or shared file system), ***run [integration test create_object_h2_spec.rb](https://github.com/sul-dlss/infrastructure-integration-test)*** and/or test manually in [stage|qa] environment, in addition to specs. ⚡

Local, Amy

image

### CODE REVIEW

**Comment by justinlittman** [comment]
@amyehodge approved.

File: app/models/work_type.rb [hunk 1]
```diff
@@ -63,6 +61,8 @@ class InvalidType < StandardError; end
     'White paper', 'Working paper'
   ].freeze
 
+  MIXED_TYPES = (%w[Data Image Portfolio Music Software/Code Sound Text Video] + MORE_TYPES).uniq.sort.freeze
```
Comment (jcoyne) [inline app/models/work_type.rb:64]: Can we remove "Data", "Text" and "Video" from this list since they are already in `MORE_TYPES`?

**Review by jcoyne — CHANGES_REQUESTED** [review]

File: app/models/work_type.rb (hunk 1 above)
Comment (jcoyne) [reply app/models/work_type.rb:64]: The screenshot doesn't display "Sound", should it?

Comment (justinlittman) [reply app/models/work_type.rb:64]: Removing dupes and `uniq`. Screenshot is from earlier code.

**Review by jcoyne — APPROVED** [review]

**Review by jcoyne — CHANGES_REQUESTED** [review]
Waiting to hear back from @arcadiafalcone about similar types:
Software/code ~ Software
Sound ~ Sound recording
Video ~ Video recording

File: app/components/works/work_type_modal_component.html.erb [hunk 2]
```diff
@@ -65,8 +65,8 @@
 
           <div class="mb-4 subtype-container" data-work-type-target="moreTypes" hidden></div>
 
-          <button type="submit" 
-            class="btn btn-primary mx-auto d-block" 
+          <button type="submit"
+            class="btn btn-primary mx-auto d-block"
```
Comment (mjgiarlo) [inline app/components/works/work_type_modal_component.html.erb:69]: Is this a whitespace-only change?

Comment (peetucket) [reply app/components/works/work_type_modal_component.html.erb:69]: looks like it, possibly came from the rebase I just did?

looks like the a cypress test is failing, and may need some attention too (assuming this PR is still relevant after 9 months)

**Comment by amyehodge** [comment]
I'm not sure what the whitespace issue or the similar terms are about, but I definitely do want the change to the instructions and the display to change to show all these right away without needing the user to expand the "More options" text. I'm not clear what is in the PR.

**Comment by arcadiafalcone** [comment]
Looks like this was hashed out on Slack but never made its way back here -- when the user chooses Mixed Materials as the type, do not include the following options in the subtypes:
“Sound” (keeping "Sound recording")
“Video” (keeping "Video recording")
“Software/Code” (keeping “Software” and “Code)

From this Aug. 2022 thread:
https://stanfordlib.slack.com/archives/C69UZCJ8M/p1661786289068899?thread_ts=1661534612.213819&cid=C69UZCJ8M

File: cypress/spec/work_type_modal.cy.js [hunk 3]
```diff
@@ -62,7 +62,7 @@ describe('Work type modal', () => {
       cy.get('div.subtype-container[data-work-type-target="subtype"] div.subtype-item input').last().click()
       cy.get('button[type="submit"]').click()
     })
-    cy.url().should('include', `/collections/${collection_id}/works/new?work_type=mixed+material&subtype%5B%5D=Data&subtype%5B%5D=Video`)
+    cy.url().should('include', `/collections/${collection_id}/works/new?work_type=mixed+material&subtype%5B%5D=3D+model&subtype%5B%5D=Working+paper`)
   })
```
Comment (peetucket) [inline cypress/spec/work_type_modal.cy.js:66]: the first and last subtypes selected are slightly different now, and the test needs to reflect this

File: spec/validators/work_subtype_validator_spec.rb [hunk 4]
```diff
@@ -54,7 +54,7 @@
     end
 
     context 'with two valid subtypes and one or more valid "more" types' do
-      let(:value) { %w[Data Sound Preprint] }
+      let(:value) { %w[Data Speech Preprint] }
```
Comment (peetucket) [inline spec/validators/work_subtype_validator_spec.rb:58]: `Sound` is no longer a valid subtype for mixed media, just replace it with a different one that is valid

**Comment by peetucket** [comment]
I made some changes to the PR to include the results of the discussion above, adjusted a test, and rebased from main.  This is now the current list of subtypes shown when you select `Mixed Materials` type.  If correct, this PR is ready for a review and merge:

Screen Shot 2023-06-13 at 11 35 45 AM

**Comment by amyehodge** [comment]
@peetucket  "Music" appears to be missing. Otherwise, looks good.

**Comment by peetucket** [comment]
> @peetucket "Music" appears to be missing. Otherwise, looks good.

Good catch, thanks.  I added it back in.

**Comment by mjgiarlo** [comment]
@peetucket what's the status of this PR? Does it just need review + a conflict resolution?

**Review by mjgiarlo — APPROVED** [review]

**Pull request merged**